### DISCUSSION TITLE
queryCache.findAll not working as expected

### DISCUSSION THREAD

**Discussion by kobe651jp** [TanStack/query#4576] [Q&A]
I'm trying to get a cache of matches by partial queryKeys.

Before getting it, the cache of ['element','A'] ,['element','B'] has been acquired with useQuery.

code as follows
```
const queryCache = new QueryCache({
  onError: (error) => {
    console.log(error);
  },
  onSuccess: (data) => {
    console.log(data);
  },
});

const queries = queryCache.findAll({
  queryKey: ['element'],
});

console.log('queries', queries);
```
But always return []

I am referring to the following document: 
https://tanstack.com/query/v4/docs/reference/QueryCache.

**Comment by TkDodo** [comment]
show a codesandbox reproduction please.

**Reply by kobe651jp** [reply]
It's a little difficult to reproduce.
Isn't the basic usage wrong? If you can confirm this, I will try various things later.
If you have a sample, it would be helpful.

**Reply by TkDodo** [reply]
well the code above, as it is, creates a `new QueryCache`, which will obviously be empty. So in a real application, I would expect you to get the queryCache via `useQueryClient`...

it would also help to know what you are trying to do, because direct interaction with the queryCache like that should be rare. the `QueryClient` provides abstractions to use the cache

**Reply by kobe651jp** [reply]
Of course, use useQueryClient if I know the ID.

However, in my usage scene, I want to retrieve information once read on demand.
As far as I know, the information once read is in the cache, so I want to get that cache information. 
When saving, it is saved as ['element', ID], so I want to get all the information that starts with ['element'].

**Reply by TkDodo** [reply]
I don't know why it's so hard to create a reproduction ... I forked the `basic` example and added a button that retrieves things, once via `queryClient.getQueriesData` and once via `queryClient.getQueryCache().findAll()`

https://codesandbox.io/s/sleepy-meadow-rocmio?file=/src/index.jsx

it just works

**Reply by kobe651jp** [reply]
Thank you for reply and the sample.

It's working perfectly.The data is what I looking for.

![image](https://user-images.githubusercontent.com/44263972/204931795-cc1c16c6-eb89-42ca-9c1a-588cd6215a26.png)


By the way, queryCache.findAll still returns empty

```
            const queryCache = new QueryCache({
              onError: (error) => {
                console.log(error);
              },
              onSuccess: (data) => {
                console.log(data);
              }
            });
            const queries = queryCache.findAll({
              queryKey: ["posts"]
            });

            console.log("queries", queries);
```
https://codesandbox.io/s/boring-swanson-epv4i0?file=/src/index.jsx

**Reply by TkDodo** [reply]
> By the way, queryCache.findAll still returns empty

ahm, of course it does, because you create a `new QueryCache`, and that creates a, well, new cache that is empty...

`queryClient.getQueryCache().findAll` is also part of my reproduction, and it works fine there...
